src/WebJobs.Script.WebHost/Health/TokenExpirationService.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Properties;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Health
+{
+    internal class TokenExpirationService : BackgroundService
```

Comment:
One last thing: can you add a comment here saying this is a good health check candidate if we ever start using `Microsoft.Extensions.Diagnostics.HealthChecks`.